app/routes/goal_routes.py

```diff
@@ -0,0 +1,95 @@
+from flask import Blueprint, jsonify, make_response, request, abort
+from flask.helpers import make_response
+from flask.json import tojson_filter
+from flask.signals import request_tearing_down
+from werkzeug.utils import header_property
+from app.models.goal import Goal
+from app import db
+from datetime import datetime
+from app.routes.utils import valid_int
+from app.models.task import Task
+
+
+goal_bp = Blueprint("goal", __name__,url_prefix ="/goals")
+
+# Helper Functions
+def get_goal_from_id(goal_id):
+    valid_int(goal_id, "goal_id")
+    return Goal.query.get_or_404(goal_id, description="{goal not found}")
```

Comment:
Since `get_or_404` returns HTML and not JSON by default you might want like this instead:

```suggestion
def get_goal_from_id(goal_id):
    valid_int(goal_id, "goal_id")
    goal = Goal.query.get(goal_id)
    if goal:
        return goal
    else:
        abort(make_response({"description": "goal not found"}, 404))
```